api/controllers/user.go

```diff
@@ -184,3 +184,104 @@ func HandleNewCodeValidation(c *gin.Context) {
 
 	c.IndentedJSON(http.StatusOK, gin.H{"message": "New Code created and sended"})
 }
+
+// controller to create and send a code to reset the password
+func HandleCodeResetPassword(c *gin.Context) {
+	var form interfaces.EmailForm
+	if err := c.BindJSON(&form); err != nil {
+		fmt.Println(err)
+		c.AbortWithStatusJSON(http.StatusBadRequest, gin.H{"message": "Bad request"})
+		return
+	}
+	//Check if there is no email
+	if form.Email == "" {
+		c.AbortWithStatusJSON(http.StatusBadRequest, gin.H{"message": "Email cannot be empty"})
+		return
+	}
+
+	_, err := models.GetUserByEmail(form.Email)
+	if err != nil {
+		c.AbortWithStatusJSON(http.StatusNotFound, gin.H{"message": "This Email has not been registered"})
+		return
+	}
+	//generate code
+	resetPasswordCode := utils.GetValidationCode()
+
+	//update in database reset password code
+	err = models.UpdateResetPassCode(form.Email, resetPasswordCode)
+	if err != nil {
+		fmt.Println(err)
+		c.AbortWithStatusJSON(http.StatusInternalServerError, gin.H{"message": "Internal server error"})
+		return
+	}
+
+	//send mail with code to help reset password
+	err = utils.SendEmail(form.Email, "Here is your validation code, to reset your password", resetPasswordCode)
+	if err != nil {
+		c.AbortWithStatusJSON(http.StatusInternalServerError, gin.H{"message": "Internal server error"})
+		return
+	}
+
+	c.IndentedJSON(http.StatusOK, gin.H{"message": "New Code, to reset password, created and sended"})
```

Comment:
Responses should follow this structure: 

```json
{
  "error": "boolean",
  "message": "String", 
  ... 
}
```

Please add the `error` field, at this way we can start standardize our API.